flowtools/convert_fcstxt_to_sce/FCStxtConvertSCE.R

```diff
@@ -0,0 +1,227 @@
+                                 grep(fl_cols_assay, pattern="SSC"),
+                                 grep(fl_cols_assay, pattern="FSC-A"),
+                                 grep(fl_cols_assay, pattern="SSC-A"),
+                                 grep(fl_cols_assay, pattern="FSC-W"),
+                                 grep(fl_cols_assay, pattern="SSC-W"),
+                                 grep(fl_cols_assay, pattern="FSC-H"),
+                                 grep(fl_cols_assay, pattern="SSC-H"),
+                                 grep(fl_cols_assay, pattern="Time", ignore.case = T),
+                                 grep(fl_cols_assay, pattern="Population|flowSOM|cluster|SOM|pop|cluster", ignore.case = T),
+                                 grep(fl_cols_assay, pattern="Live_Dead|live|dead", ignore.case = T))
+
+        fl_cols_assay <- fl_cols_assay[-channels_to_exclude]
+    }
+    counts <- flowtext[, fl_cols_assay, drop = FALSE]
+    counts <- as.matrix(counts)
+
+    # transpose data into assay as columns=cells and rows=features.
+    counts <- base::t(counts)
+    colnames(counts) <- 1:ncol(counts)
+
+
+    #-----------------#
+    #coldata/meta data#
+    #-----------------#
+
+    # by default any columns with sample names or cluster results will be extracted - to over ride this user must provide a comma separated list of column name (mtd_cols)
+    mtd_cols_assay <- colnames(flowtext)
+    if (length(mtd_cols) > 0){
+        if(length(mtd_cols) > ncol(flowtext)){
+            quit(save = "no", status = 14, runLast = FALSE)
+        }
+        mtd_cols_assay <- mtd_cols_assay[mtd_cols_assay %in% mtd_cols]
+    } else {
+        #print("Meta data columns from flowtext files not specified")
+        #create warning here to the user - but without failing
+        mtd_columns <- c(grep(marker_type, pattern="sample", ignore.case=T),
+                         grep(marker_type, pattern="population|flowsom|cluster|pop|som", ignore.case=T))
+
+        mtd_cols_assay <- mtd_cols_assay[mtd_columns]
+    }
+
+    md <- flowtext[, mtd_cols_assay, drop = FALSE]
+
+    # if metadata available will be merged with meta data from flow text
+    if(!is.null(meta_data)){
+
+        #match column names so case insensitive
+        md_col <- tolower(colnames(md))
+        mtd_col <- tolower(colnames(meta_data))
+
+        #quit if < 1 or > 1 column names match
+        if(length(intersect(md_col, mtd_col)) == 0){
+            quit(save = "no", status = 15, runLast = FALSE)
+        }
+        if(length(intersect(md_col, mtd_col)) > 1){
+            quit(save = "no", status = 16, runLast = FALSE)
+        }
+
+        #merge by matched column
+        meta_data <- merge(x = md, y = meta_data, all=T)
+
+    }
+
+    #create Single Cell experiment object. SCOPE requires both counts and logcounts assays - for FLOW both assays contain the same data
+    sce <- SingleCellExperiment(assays = list(counts=counts, logcounts=counts))
```

Comment:
@emily101-gif @twfs I'm not very at ease with filling logcounts with counts, this could generate errors later on as anyone using the rds object later on will assume that the content of logcounts is actually log counts and not counts. Is there are good reason for doing this? otherwise, if we need log counts, I would suggest that we compute them properly instead of storing counts there.

Reply:
I guess that if it is usual that in FLOW counts and logcounts are left with the same data, then we could allow this to be left like this for compatibility with SCOPE.